pkg/oci/oci.go

```diff
@@ -180,15 +177,31 @@ func pullIfNotExist(ctx context.Context, imageStore oras.Target, authOpts *AuthO
 		return fmt.Errorf("resolving image %q: %w", image, err)
 	}
 
+	_, err = pullImage(ctx, targetImage, imageStore, authOpts)
+	if err != nil {
+		return fmt.Errorf("pulling image %q: %w", image, err)
+	}
+
+	return nil
+}
+
+func pullImage(ctx context.Context, targetImage reference.Named, imageStore oras.Target, authOpts *AuthOptions) (*ocispec.Descriptor, error) {
+	if authOpts.DisallowPulling {
+		return nil, errors.New("pulling is disabled")
```

Comment:
```suggestion
		return nil, errors.New("pulling is disallowed")
```

(sorry about being a jerk, but better to align the naming as good as possible)